src/service/history-impl.js

```diff
@@ -109,6 +112,20 @@ export class History {
   }
 
   /**
+   *
+   * @param {string} target
+   * @param {string} previousTarget
```

Comment:
I was trying to avoid re-parsing URL but I am ok with getting it from win.location though in here.